app/assets/javascripts/initializers/initializeCommentsPage.js.erb

```diff
@@ -3,74 +3,82 @@ var iconSmallThread = `<svg width="24" height="24" class="crayons-icon" viewBox=
 function initializeCommentsPage() {
   if (document.getElementById('comments-container')) {
     toggleCodeOfConduct();
+    var userStatus = document.body.getAttribute('data-user-status');
     var commentableId = document.getElementById('comments-container').dataset.commentableId;
     var commentableType = document.getElementById('comments-container').dataset.commentableType;
+    var hasRecentCommentActivity = document.getElementById('comments-container').dataset.hasRecentCommentActivity;
     commentableIdList = commentableId.split(",");
-    var f = (function() {
-      for (var i = 0; i < commentableIdList.length; i++) {
-        (function(i){
-          var ajaxReq;
-          if (window.XMLHttpRequest) {
-            ajaxReq = new XMLHttpRequest();
-          } else {
-            ajaxReq = new ActiveXObject('Microsoft.XMLHTTP');
-          }
-          ajaxReq.onreadystatechange = function () {
-            if (ajaxReq.readyState === XMLHttpRequest.DONE) {
-              var responseObj = JSON.parse(ajaxReq.response);
-              var reactions = responseObj.reactions;
-              var allNodes = document.getElementsByClassName('single-comment-node');
-              var publicReactionCounts = responseObj.public_reaction_counts;
-              for (var i = 0; i < reactions.length; i++) {
-                var buttForComment = document.getElementById('button-for-comment-' + reactions[i].reactable_id);
-                if (buttForComment) {
-                  buttForComment.classList.add('reacted');
+
+    // When no logged in user only need to fetch reaction numbers if there has been recent comment activity.
+    // If no recent comments, we do not need the precision of fetching latest counts async. Static value is almost certainly good enough.
+    // Always fetch when user is logged in because we need state of whether the user has reacted.
```

Comment:
Since we're in this code and adding some comments...can we use JSDoc here instead? 😃 This function has a lot going on, and definitely could use some documentation. Check out the `initializeBroadcast.js` [file](https://github.com/forem/forem/blob/master/app/assets/javascripts/initializers/initializeBroadcast.js ) which has some helpful examples that you can follow :)

I think our future selves will really appreciate it 🙏🏽

Reply:
Thanks for this! I will definitely make much better use of jsdoc in the future.